src/System.ServiceModel.Tests.Common/src/testproperties.targets

```diff
@@ -0,0 +1,167 @@
+	      sb.AppendLine("{");
+	      sb.AppendLine("    public partial class TestProperties");
+	      sb.AppendLine("    {");
+        
+        // Generate a string for each property name.
+        // Code accessing the properties should use these generated names.
+        foreach (var item in TestPropertyItems)
+        {
+            sb.AppendLine(String.Format("        public static readonly string {0}_PropertyName = \"{0}\";", item.ItemSpec));
+        }
+        
+        // Generate the Initialize partial method to populate the dictionary
+        sb.AppendLine();
+	      sb.AppendLine("        partial void Initialize(Dictionary<string, string> properties)");
+	      sb.AppendLine("        {");
+                
+        foreach (var item in TestPropertyItems)
+        {
+            string propertyName = item.ItemSpec;
+            string propertyValue = item.GetMetadata("Value");
+            if (String.IsNullOrEmpty(propertyValue))
+            {
+                propertyValue = item.GetMetadata("DefaultValue");
+            }
+         
+            // Convert the property value into a legal string literal.     
+            StringBuilder literal = new StringBuilder();       
+            foreach (var c in propertyValue)
+            {
+                switch (c)
+                {
+                    case '\'': literal.Append(@"\'"); break;
+                    case '\"': literal.Append("\\\""); break;
+                    case '\\': literal.Append(@"\\"); break;
+                    case '\0': literal.Append(@"\0"); break;
+                    case '\a': literal.Append(@"\a"); break;
+                    case '\b': literal.Append(@"\b"); break;
+                    case '\f': literal.Append(@"\f"); break;
+                    case '\n': literal.Append(@"\n"); break;
+                    case '\r': literal.Append(@"\r"); break;
+                    case '\t': literal.Append(@"\t"); break;
+                    case '\v': literal.Append(@"\v"); break;
+                    default:
+                        if (c >= 0x20 && c <= 0x7e)
+                        {
+                            // ASCII printable character
+                            literal.Append(c);
+                        }
+                        else
+                        {
+                            // As UTF16 escaped character
+                            literal.Append(@"\u");
+                            literal.Append(((int)c).ToString("x4"));
+                        }
+                        break;
+                }
+            }
+            
+            string setStatement = String.Format("            properties[\"{0}\"] = \"{1}\";", propertyName, literal.ToString());
+            sb.AppendLine(setStatement);
+        }
+            
+	      sb.AppendLine("        }");
+	      sb.AppendLine("    }");
+	      sb.AppendLine("}");
+        
+        // Do not rewrite file if there are no changes.  This permits incremental builds.
```

Comment:
I did move the property item collection and property name out to a separate .props file so that the .targets file contains only the Task.  But I don't believe it is possible to make MsBuild understand when the output of the CreateTestProperties target would be up to date.  Its inputs come from both the .props file but also from MSBuild properties.  If I set the environment variable UseFiddlerUrl to true and rebuilt, it would be necessary to regenerate the file.